winter/core/utils/typing.py

```diff
@@ -37,6 +38,12 @@ def get_union_args(type_: object) -> list:
 
 
 def get_origin_type(hint_class):
+    if hasattr(types, 'UnionType') and isinstance(hint_class, types.UnionType):
+        # Extract the arguments of the union (e.g., `str | int` -> (str, int))
+        args = get_args(hint_class)
+        # Convert to the old `typing.Union` style
```

Comment:
To simplify future life, usually it's better to rely on newer style, not vice-versa.